### ISSUE TITLE
state of javascript/react

### ISSUE THREAD

**Issue by axyz** [syl20bnr/spacemacs#5848]
I'm getting really annoyed by js2-mode syntax checker, I cannot find a way to really disable it and while it plays somewhat decently together with eslint it is starting to fail very badly when working with some es7 such as object spreads and async/await.

I'm planning to create a branch for the react-layer where I do not use anymore the js2-minor. Currently web-mode is really improved and eslint is the de-facto standard for linting (especially now that they are merging with jscs).

I do not use a lot of js2 functionalities such as refactor and similar. I think that having web-mode tailored for jsx and js with tern and maybe skewer if it could play nicely with babel code should be enough for me.

To note that without js2-minor on react layer also auto-completion is working much better.

My question to current react-layer user is what js2-mode features they will miss if there are any.

And also I'm more widely asking for alternative solutions to only disable the js2 syntax checker.

I think react-layer may be then also used for regular js code and provide some insights on how a web-mode based layer tailored for js may work to see if would make sense in the future to think about creating a new js layer based on web-mode or if it would be better to try to fix js2-mode.

**Comment by TheBB** [comment]
I've closed #4735 in favor of this. Please go ahead. Even I, who do not write react, would love this. :-)

**Comment by syl20bnr** [comment]
The proposal has been merged, waiting for feedback.

**Comment by growlove** [comment]
If I had to choose between accurate highlighting/linting and the js2-mode features that might be lost, I'd choose the former hands down. Broken highlighting has been the biggest point of frustration for me in using spacemacs for react/redux stuff.

**Comment by axyz** [comment]
small bug discovered so far: when using tab to correctly indent some code the eslint error will not disappear until the next edit. Not sure if it is a known flycheck problem or it is specific to eslint (or the way eslint process and flycheck interact).

@syl20bnr please also consider merging #5860 as part of this topic: in lot of cases eslint may not work as expected without that fix.

I will then create a new PR to try out skewer

So far the most "annoying" thing in daily usage is multiline comments also for single lines, but a positive note is that they seems to work correctly also inside jsx code

**Comment by sdn90** [comment]
Would love to see #5850 in Master. Removing the js2 syntax checker made React development somewhat bearable for me.

**Comment by arronmabrey** [comment]
I'm with @sdn90. Looks like this has been ready in develop for a few months. Is there anything blocking it?

**Comment by kolpav** [comment]
Any progress on this? I would love to use spacemacs for react development but it is just not good at the moment and I spent most of the time solving little issues like highlighting, linting, indents etc.. instead of being productive.

**Comment by d12frosted** [comment]
Fixed with released of Spacemacs v0.200. Let us know if you still have any problems with this issue.

**Comment by kolpav** [comment]
Hello, congrats on a big release!
I installed the fresh version of new spacemacs and first thing I tried didn't work as I would expect. So I want to use this issue to ask if anybody else have this problem or problem is on my side.
I have simple component
```javascript
const Foo = () =>
  <div>
    <Bar
      color='red' // <- I have cursor here and press `o`
    />
  </div>

const Foo = () =>
  <div>
    <Bar
      color='red'
  // cursor jumps here
    />
  </div>

const Foo = () =>
  <div>
    <Bar
      color='red'
      // Instead of here (pressing tab does nothing)
    />
  </div>
```
Just for the record I am really new to the whole cool kids editor like emacs/vim so I might be missing something obvious here. Is there any others react js developers out there who have spacemacs all set up so indentation works as it should?
